src/renderer/store/selectors/channel.ts

```diff
@@ -136,31 +133,22 @@ const concatMessages = (mainMsg, messagesToConcat) => {
 
 export const mergeIntoOne = (messages: any[]) => {
   if (messages.length === 0) return
-  const result = [[]]
-  let last: any = null
+  const result = []
+  const timeOfStackMessages = 300 // in seconds
+  let lastPubKey = null
+  let lastCteatedAt = null
   for (const msg of messages) {
-    const isMessageInTargetZone = last
-      ? checkMessageTargetTimeWindow({
-        targetCreatedAt: last.createdAt,
-        timeStamp: msg.createdAt,
-        // eslint-disable-next-line
-        timeWindow: last.createdAt + 300
-      })
-      : true
-    if (last && msg.status === 'failed') {
+    if (lastPubKey && lastPubKey !== msg.pubKey) {
       result.push([])
       result[result.length - 1].push(msg)
-    } else if (last && (msg.type !== 1 || last.type !== 1)) {
-      result.push([])
-      result[result.length - 1].push(msg)
-    } else if ((last && last.sender.nickname !== msg.sender.nickname) || !isMessageInTargetZone) {
-      result.push([])
-
+    } else if (lastCteatedAt && ((lastCteatedAt - msg.createdAt) < timeOfStackMessages)) {
       result[result.length - 1].push(msg)
     } else {
+      result.push([])
       result[result.length - 1].push(msg)
     }
-    last = msg
+    lastPubKey = msg.pubKey
+    lastCteatedAt = msg.createdAt
```

Comment:
Does this approach allow us to style grouped messages differently than, say, a single message with a bunch of line breaks in it? 

Will it let us put controls on individual messages, say, if you wanted to delete a single message within a series of grouped messages?